### PR TITLE
Functionality to easily add new metrics

### PR DESCRIPTION
**Pull request by grokys**
Updates the metrics reporting:
- Store the metrics in a different format, with separate Measures and Dimensions
- Store reports for each day separately
- Change the serialization functionality to serialize all dimensions as strings

Depends on #1130

### CODE REVIEW

File: src/GitHub.VisualStudio/Services/UsageService.cs [hunk 1]
```diff
@@ -79,9 +113,9 @@ public IDisposable StartTimer(Func<Task> callback, TimeSpan dueTime, TimeSpan pe
                 var json = SimpleJson.SerializeObject(data);
```
Comment (grokys) [inline src/GitHub.VisualStudio/Services/UsageService.cs]: Note that this is going to serialize the fields with PascalCasing. I guess we need to write a `CamelCaseSerializerStrategy` if we want to serialize it using camelCase.

Comment (shana) [reply src/GitHub.VisualStudio/Services/UsageService.cs]: It doesn't matter much here, the method that post the data serializes the model into camel case.

File: src/GitHub.Exports/Models/UsageModel.cs [hunk 2]
```diff
@@ -1,65 +1,61 @@
...
-            foreach (var property in properties)
+            return new UsageModel
             {
-                var cloneValue = property.PropertyType == typeof(int);
-
-                if (property.Name == nameof(result.NumberOfStartupsWeek))
-                    cloneValue = includeWeekly;
-                else if (property.Name == nameof(result.NumberOfStartupsMonth))
-                    cloneValue = includeMonthly;
+                Guid = guid,
```
Comment (shana) [inline src/GitHub.Exports/Models/UsageModel.cs]: I think the GUID needs to be in the dimensions model and not here.

**Comment by jcansdale** [comment]
Was there a plan to somehow exclude metrics from our development usage? Would we turn off metrics for debug builds or include an `IsDebugBuild` field in the metrics?

**Comment by grokys** [comment]
@jcansdale sending metrics is already disabled in debug builds.

**Comment by StanleyGoldman** [comment]
https://github.com/github/central/pull/354

This WIP pull request will absorb the changes on the receiving side.

File: src/GitHub.VisualStudio/Services/UsageTrackerDispatcher.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+using Microsoft.VisualStudio.Shell;
+using System.Threading.Tasks;
+using GitHub.Models;
+using System.Linq.Expressions;
+
+namespace GitHub.Services
+{
+    [ExportForProcess(typeof(IUsageTracker), "devenv")]
+    [PartCreationPolicy(CreationPolicy.NonShared)]
+    public class UsageTrackerDispatcher : IUsageTracker
```
[the remark this answers is not in the document]
Comment (StanleyGoldman) [reply src/GitHub.VisualStudio/Services/UsageTrackerDispatcher.cs]: It wasn't even included in the project file.

**Pull request merged**